django/contrib/admin/models.py

```diff
@@ -44,6 +44,27 @@ def log_action(
             change_message=change_message,
         )
 
+    def log_actions(self, user_id, queryset, action_flag, change_message=""):
+        """
+        Method to bulk_create multiple LogEntry objects
+        from a given queryset.
+        """
+        if isinstance(change_message, list):
+            change_message = json.dumps(change_message)
+        log_entry_list = [
+            self.model(
+                user_id=user_id,
+                content_type_id=ContentType.objects.get_for_model(obj).id,
+                object_id=str(obj.pk),
+                object_repr=repr(obj)[:200],
+                action_flag=action_flag,
+                change_message=change_message,
+            )
+            for obj in queryset
+        ]
+
+        return self.model.objects.bulk_create(log_entry_list)
```

Comment:
This will not work on all supported databases, because `bulk_create()` doesn't return IDs on SQLite < 3.35, MariaDB < 10.5, Oracle, and probably on most of databases supported via 3rd party backends (as [documented](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create)).